### DISCUSSION TITLE
how to access elements in View from ViewModel

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#15723] [Q&A]
I have a TextBox and in the ViewModel file I need to get text from this TextBox but I can’t access graphic elements from the View. How can I do it?

**Comment by timunie** [comment]
Use a Binding?

**Reply by al-zakaria** [reply]
so this is the question: I don’t know how to implement this Binding and ChatGPT does not provide any useful information on this issue

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia.Samples?tab=readme-ov-file#mvvm-samples 

check out our MVVM samples please.

**Reply by al-zakaria** [reply]
Sorry but as I looked there only Reactive UI and not Community Toolkit(((

**Reply by thevortexcloud** [reply]
The binding process is the same in Community Toolkit. The only real difference is how you declare your properties. I strongly suggest reading some tutorials instead of relying on ChatGPT. Even WPF tutorials will work just as well here as WPF's binding system is pretty much exactly the same as Avalonia's.

https://docs.avaloniaui.net/docs/basics/data/data-binding/

You don't even need any third party libraries for this, as long as the view model implements `INotifyPropertyChanged` and raises change events properly.

**Comment by Pratham95VP** [comment]
Check this.   https://docs.avaloniaui.net/docs/guides/data-binding/inotifypropertychanged
